atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/SweepStrategyManagers.java

```diff
@@ -33,13 +35,15 @@ private SweepStrategyManagers() {
     }
 
     public static SweepStrategyManager createDefault(KeyValueService kvs) {
-        RecomputingSupplier<Map<TableReference, SweepStrategy>> supplier =
-                RecomputingSupplier.create(() -> getSweepStrategies(kvs));
+        // On a cache miss, load metadata only for the relevant table. Helpful when many dynamic tables.
+        LoadingCache<TableReference, SweepStrategy> cache = Caffeine.newBuilder()
+                .build(tableRef -> getSweepStrategy(kvs.getMetadataForTable(tableRef)));
         return tableRef -> {
-            SweepStrategy strategy = supplier.get().get(tableRef);
-            if (strategy == null) {
-                strategy = supplier.recompute().get(tableRef);
+            // Add all existing tables the first time this is called, to optimize for cases when mostly non-dynamic tables.
+            if (cache.estimatedSize() == 0) {
+                cache.putAll(getSweepStrategies(kvs));
             }
+            SweepStrategy strategy = cache.get(tableRef);
```

Comment:
The way you wrote this, this will never return null.

Reply:
It could be if we queried for a tableRef for a table that had already been deleted by the time Atlas initialized, right?

Reply:
`getSweepStrategy` never returns `null` though right? Similarly `TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(tableMeta).getSweepStrategy()`.

Reply:
Oh yeah.. argh, I need coffee before I deal with support tickets in the morning